### ISSUE TITLE
Fails to build client when cluster has no CA bundle set

### ISSUE THREAD

**Issue by ctron** [clux/kube-rs#164]
With `0.27.0` (don't know when it broke): When the local kube config has no explicit CA bundle set, kube fails to load the configuration:

~~~
Error loading kube config: Error loading kube config: Failed to get data/file with base64 format
~~~

It does load the CA bundle in `create_client_builder`:
~~~rust
    for ca in loader.ca_bundle()? {
        client_builder = hacky_cert_lifetime_for_macos(client_builder, &ca);
        client_builder = client_builder.add_root_certificate(ca.try_into()?);
    }
~~~

Which at some point calls `load_certificate_authority`, which dives into `data_or_file_with_base64`. However that expects either plain data, or a file:

~~~rust
match (data, file) {
        (Some(d), _) => { ...  }
        (_, Some(f)) => { ... }
        _ => Err(Error::KubeConfig(
            "Failed to get data/file with base64 format".into(),
        )),
    }
~~~

However, if you have a trusted certificate, then there is no need to add a CA bundle. This used to work in the past also with `kube`.

**Comment by clux** [comment]
This was probably introduced during the rustls introduction. If you wanna make a PR from that commit, I'll get it out ASAP.

**Comment by clux** [comment]
Released in 0.28.0. Sorry I didn't get time to do a maintenance release of it so now you get the big update :sweat_smile: 

Thanks for the report + fix at any rate!
